### PR TITLE
Initial submission for jonathan_aotearoa.

### PR DESCRIPTION
**Pull request by jonathan-aotearoa**
#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [x] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)

* Execution time:
* Execution time of reference implementation:

### CODE REVIEW

**Comment by jonathan-aotearoa** [comment]
Hash collisions are dealt with on line 305 in the Repository findIndex method.

**Comment by jonathan-aotearoa** [comment]
Execution times on my machine:

* `./calculate_average_baseline.sh  125.47s user 6.96s system 100% cpu 2:11.29 total`
* `./calculate_average_jonathanaotearoa.sh  17.45s user 2.03s system 766% cpu 2.540 total`

**Comment by jonathan-aotearoa** [comment]
@gunnarmorling - the issue with my failed build appears to be due to my GitHub username having a hyphen in it.

**Comment by gunnarmorling** [comment]
One of the tests is failing (see CI for details).

**Comment by jonathan-aotearoa** [comment]
> One of the tests is failing (see CI for details).

It looks like it was passing locally for me because it wasn't failing if a segfault was detected.

**Comment by jonathan-aotearoa** [comment]
@gunnarmorling - I couldn't reproduce the build error locally. I've made some changes in the hope of fixing whatever root cause is/was.

**Comment by jonathan-aotearoa** [comment]
@gunnarmorling - I think I see the problem now! Please wait for an additional commit before re-running the tests. Thanks!

**Comment by jonathan-aotearoa** [comment]
@gunnarmorling - It should be good to go now.

File: src/main/java/dev/morling/onebrc/CalculateAverage_jonathanaotearoa.java [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+            }
+        }
+
+        public Stream<StationData> entries() {
+            return Arrays.stream(table).filter(Objects::nonNull);
+        }
+
+        private int findIndex(int nameHash) {
+            int i = (nameHash & 0x7FFFFFFF) % table.length;
+            while (table[i] != null && table[i].nameHash != nameHash) {
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_jonathanaotearoa.java:420]: I don't think comparign the hash is enough, you'd have to compare the actual name.

Comment (jonathan-aotearoa) [reply src/main/java/dev/morling/onebrc/CalculateAverage_jonathanaotearoa.java]: Good point! A bit of an oversight on my behalf ;)

**Comment by jonathan-aotearoa** [comment]
@gunnarmorling  - Updated the collision check to include checking the value of both names in memory. See line 484.

**Comment by gunnarmorling** [comment]
Yepp, this looks better now. Results are still off for the 10K key set test though (see create_measurements3.sh):

```
Validating calculate_average_jonathan-aotearoa.sh -- measurements_10K_1B.txt
Using native image 'target/CalculateAverage_jonathan-aotearoa_image'. Delete this file to select JVM mode.
88c88
< Alco;-15.0;14.6;46.2
---
> Alco;-14.5;17.3;47.1
94c94
< Alot;-10.2;18.1;48.6
---
> Alot;-12.1;17.1;48.5
108c108
< Anj;-19.9;14.5;47.7
---
> Anj;-19.9;13.2;41.2
142c142
< Bag;-13.6;14.7;49.0
---
> Bag;-14.0;14.1;43.7
150c150
< Bane;-20.5;12.2;41.1
---
> Bane;-24.1;10.6;41.6
170c170
< Ben;-17.5;16.7;49.1
---
> Ben;-10.9;18.8;49.1
172c172
< Bent;-23.5;10.9;43.0
---
> Bent;-14.4;14.8;46.4
195c195
< Bnj;-12.4;17.6;53.8
---
> Bnj;-11.1;19.3;53.8
234c234
< Bālu;-18.3;15.0;50.9
---
> Bālu;-8.4;19.8;52.0
245c245
< Caku;-21.2;16.2;52.5
---
> Caku;-12.4;17.6;52.5
270c270
< Cen;-21.0;11.5;46.7
---
> Cen;-21.0;8.4;38.4
379c379
< El P;-14.5;18.1;51.2
---
> El P;-11.8;19.3;51.2
405c405
< Fita;-19.0;10.2;39.7
---
> Fita;-19.0;11.7;41.0
583c583
< Juiz;-15.6;17.6;49.9
---
> Juiz;-15.6;16.0;49.9
650,651c650,651
< Kyan;-16.6;16.7;47.2
< Kyau;-19.8;11.1;41.9
---
> Kyan;-16.6;17.1;47.2
> Kyau;-18.6;12.5;41.8
703c703
< Limo;-13.9;16.8;47.7
---
> Limo;-10.9;20.0;47.7
706c706
< LiraCarcarNegage;-15.8;17.1;51.3
---
> LiraCarcarNegage;-12.1;18.4;51.3
726c726
< Lucé;-23.4;12.9;51.0
---
> Lucé;-9.6;19.0;51.0
738c738
< Ma;-17.9;11.2;39.2
---
> Ma;-19.4;10.8;39.2
769c769
< Mapl;-13.2;17.8;47.5
---
> Mapl;-16.5;15.8;45.0
802c802
< Mich;-20.1;11.0;44.0
---
> Mich;-20.1;9.9;40.4
840c840
< Na;-22.6;11.7;43.0
---
> Na;-22.6;12.2;43.0
923c923
< Pape;-22.1;13.0;45.1
---
> Pape;-22.1;13.7;45.1
982c982
< Pālg;-23.5;15.0;52.0
---
> Pālg;-23.5;10.2;41.0
1035c1035
< SFn;-19.1;13.4;47.7
---
> SFn;-19.1;8.6;36.6
1097c1097
< Shcn;-15.3;16.8;49.0
---
> Shcn;-13.6;18.7;49.0
1191c1191
< Tees;-18.1;12.3;44.4
---
> Tees;-20.1;8.6;38.3
1345c1345
< a Al;-13.6;16.8;46.7
---
> a Al;-15.3;14.9;46.0
1423c1423
< aCa;-15.7;15.5;48.0
---
> aCa;-15.7;15.0;46.3
1451c1451
< aFon;-17.1;14.7;47.1
---
> aFon;-17.1;11.9;42.7
1486c1486
< aJi;-16.9;14.8;44.8
---
> aJi;-16.9;14.8;46.0
1621c1621
< aVic;-23.5;9.9;43.8
---
> aVic;-16.5;13.6;43.8
1657c1657
< aay;-21.6;15.7;46.8
---
> aay;-21.6;14.0;47.0
1683c1683
< abui;-17.0;12.0;46.1
---
> abui;-16.2;11.0;41.1
1712c1712
< acres;-21.0;15.9;50.5
---
> acres;-13.2;16.7;50.5
1730c1730
< adaB;-8.6;19.8;51.3
---
> adaB;-8.6;19.9;47.9
1787c1787
< agen;-12.4;18.0;48.1
---
> agen;-10.8;19.3;48.1
1831c1831
< ahā;-19.4;14.2;45.0
---
> ahā;-18.7;16.5;45.0
1834c1834
< ai-t;-16.7;14.2;44.9
---
> ai-t;-18.1;12.5;44.9
1937c1937
< ala;-11.7;18.2;45.6
---
> ala;-11.4;18.5;46.0
2110c2110
< anH;-12.6;19.8;51.0
---
> anH;-12.0;19.9;51.0
2139c2139
< anal;-20.3;10.2;41.0
---
> anal;-20.3;8.7;38.2
2195c2195
< ango;-16.3;15.4;45.4
---
> ango;-16.3;15.5;45.4
2207c2207
< anim;-16.8;13.8;42.4
---
> anim;-18.5;12.8;43.2
2247c2247
< ante;-8.2;19.3;51.5
---
> ante;-7.6;20.0;51.5
2273c2273
< aoDj;-13.0;18.3;47.7
---
> aoDj;-11.6;19.4;48.0
2296c2296
< apat;-16.5;15.4;49.4
---
> apat;-8.5;19.9;50.1
2595c2595
< avis;-21.2;10.9;46.4
---
> avis;-23.5;7.0;36.6
2603c2603
< awKi;-13.8;16.9;47.5
---
> awKi;-16.7;12.9;44.7
2692c2692
< bCa;-11.3;17.3;46.3
---
> bCa;-10.9;18.1;48.0
2762c2762
< bla;-12.7;17.7;51.8
---
> bla;-12.7;17.4;51.8
2836c2836
< cJi;-13.8;16.2;46.0
---
> cJi;-13.8;17.0;45.4
2844c2844
< cSa;-11.2;19.1;46.9
---
> cSa;-11.2;19.1;49.9
2890c2890
< chal;-11.2;19.0;48.9
---
> chal;-11.2;19.1;55.1
3006c3006
< dDan;-15.6;13.0;41.9
---
> dDan;-15.7;12.4;41.7
3012c3012
< dJi;-13.1;15.7;45.3
---
> dJi;-15.3;15.2;44.6
3025,3026c3025,3026
< dS;-20.1;11.3;48.4
< dSa;-22.6;11.1;49.9
---
> dS;-20.1;8.7;37.6
> dSa;-22.6;6.3;35.9
3078c3078
< dd;-17.1;13.0;42.0
---
> dd;-16.4;14.2;44.7
3103c3103
< dha;-18.8;13.3;45.3
---
> dha;-18.8;12.2;41.4
3113c3113
< diXi;-18.6;10.7;48.2
---
> diXi;-22.3;8.7;37.3
3317c3317
< eMyt;-18.7;14.2;44.2
---
> eMyt;-19.4;11.9;41.3
3327c3327
< ePoi;-16.5;17.8;47.4
---
> ePoi;-10.2;19.7;47.5
3339,3340c3339,3340
< eS;-19.0;14.1;49.4
< eSa;-20.0;18.2;47.9
---
> eS;-11.6;17.6;49.4
> eSa;-9.6;19.8;47.9
3405c3405
< ed;-22.1;10.9;44.7
---
> ed;-22.1;9.7;38.1
3408c3408
< edan;-14.2;16.8;51.1
---
> edan;-21.8;14.0;42.8
3466c3466
< eha;-16.7;17.0;50.9
---
> eha;-8.2;19.0;50.9
3488c3488
< eiy;-9.8;19.9;53.7
---
> eiy;-10.4;20.0;53.7
3546c3546
< elsK;-12.0;18.6;48.3
---
> elsK;-9.0;18.7;51.9
3615,3616c3615,3616
< enge;-15.0;16.6;49.8
< engl;-22.0;11.2;44.8
---
> enge;-15.0;16.2;49.8
> engl;-22.0;9.7;44.8
3866c3866
< eyun;-19.8;8.7;41.2
---
> eyun;-19.8;8.6;38.0
3928c3928
< fuSh;-16.8;13.4;49.0
---
> fuSh;-17.0;14.2;49.0
3954c3954
< gJi;-15.3;17.8;47.6
---
> gJi;-11.2;18.2;47.6
3968c3968
< gSou;-18.5;13.8;48.3
---
> gSou;-16.8;14.9;48.3
3976c3976
< gWol;-24.1;12.2;42.6
---
> gWol;-16.3;13.7;42.6
4158c4158
< h H;-18.7;13.5;47.2
---
> h H;-11.7;18.2;47.7
4193c4193
< hS;-21.1;9.1;41.0
---
> hS;-21.1;8.8;37.5
4257c4257
< haū;-20.0;11.7;40.8
---
> haū;-17.1;12.7;40.8
4262c4262
< he;-16.9;16.0;46.4
---
> he;-11.7;17.8;46.4
4265c4265
< heKg;-17.1;14.4;50.6
---
> heKg;-21.4;11.3;41.9
4323c4323
< hint;-12.1;18.3;48.7
---
> hint;-12.1;17.4;47.8
4343c4343
< hkot;-16.7;16.9;50.1
---
> hkot;-13.6;19.7;50.1
4367c4367
< horj;-9.4;19.8;49.3
---
> horj;-8.5;19.6;47.0
4405c4405
< huji;-14.8;14.1;46.2
---
> huji;-14.8;14.6;46.2
4438c4438
< hī;-11.7;19.8;51.0
---
> hī;-11.7;19.7;49.9
4455c4455
< i;-14.7;15.9;46.7
---
> i;-14.4;16.6;46.7
4547c4547
< iS;-15.2;17.2;48.5
---
> iS;-11.4;17.7;48.5
4569c4569
< iWuz;-22.3;10.6;41.3
---
> iWuz;-15.9;12.5;48.2
4668c4668
< idad;-17.5;10.9;42.6
---
> idad;-20.4;9.5;40.7
4689c4689
< ie;-19.7;13.7;45.5
---
> ie;-19.7;11.0;40.9
4733c4733
< ihla;-7.3;19.8;53.8
---
> ihla;-8.3;19.7;53.8
4938c4938
< ioun;-10.9;19.3;48.4
---
> ioun;-10.9;19.8;48.7
4952c4952
< ir;-21.7;8.1;40.0
---
> ir;-21.7;7.6;40.0
5031c5031
< itat;-16.6;16.8;46.5
---
> itat;-16.6;13.6;46.7
5037c5037
< itij;-21.4;14.4;47.9
---
> itij;-11.0;17.4;50.6
5066c5066
< iuji;-15.1;13.5;43.8
---
> iuji;-15.1;13.0;40.8
5122c5122
< j;-21.4;13.3;46.7
---
> j;-21.4;11.9;41.1
5153c5153
< jen;-15.1;18.1;49.6
---
> jen;-14.1;19.6;49.6
5177c5177
< k;-15.5;17.8;49.9
---
> k;-9.3;19.1;49.9
5183c5183
< kBa;-14.9;16.3;50.0
---
> kBa;-15.5;17.3;50.0
5249c5249
< ke;-18.0;13.2;45.2
---
> ke;-15.4;13.4;45.2
5255c5255
< ken;-18.5;13.1;46.3
---
> ken;-18.5;11.2;42.7
5365c5365
< lBa;-19.5;15.4;46.0
---
> lBa;-19.5;14.2;44.0
5370c5370
< lChe;-12.9;17.1;46.1
---
> lChe;-10.3;18.7;47.1
5511c5511
< le;-23.1;9.5;39.7
---
> le;-23.1;9.3;39.4
5571c5571
< lhī;-14.0;17.6;52.6
---
> lhī;-11.7;19.1;52.6
5584c5584
< liSilianaPort Ch;-15.8;16.1;48.3
---
> liSilianaPort Ch;-15.8;14.9;48.3
5682c5682
< lo B;-15.5;16.0;46.4
---
> lo B;-12.7;17.5;46.4
5713c5713
< lovi;-11.8;18.3;48.0
---
> lovi;-13.0;17.2;47.7
5788c5788
< m;-18.3;14.4;49.4
---
> m;-18.3;13.9;44.7
5819c5819
< mVi;-23.4;10.4;42.5
---
> mVi;-23.4;8.7;36.8
5835c5835
< maam;-12.5;17.6;49.4
---
> maam;-11.6;19.3;49.4
5864c5864
< mba;-16.6;15.3;45.2
---
> mba;-16.0;15.2;45.2
5917c5917
< mir;-13.0;15.4;45.1
---
> mir;-14.3;15.7;46.8
6010c6010
< n d;-24.2;11.2;44.2
---
> n d;-24.2;9.8;38.6
6024c6024
< nBa;-18.8;14.6;44.1
---
> nBa;-18.8;14.7;44.1
6034c6034
< nCan;-15.7;14.9;45.9
---
> nCan;-16.2;13.4;45.1
6156c6156
< nVi;-19.6;15.0;45.9
---
> nVi;-13.9;16.6;45.9
6182c6182
< naT;-16.6;17.3;50.6
---
> naT;-8.7;19.8;50.6
6210c6210
< nan;-16.9;13.4;46.7
---
> nan;-16.9;12.6;42.4
6227c6227
< nba;-16.0;15.3;43.3
---
> nba;-16.6;15.4;43.4
6266c6266
< ndb;-19.0;14.8;46.9
---
> ndb;-12.9;19.2;48.1
6370c6370
< ngga;-10.2;18.6;51.9
---
> ngga;-12.0;18.5;48.3
6442c6442
< nir;-16.4;13.7;46.8
---
> nir;-16.4;12.7;43.1
6467c6467
< nli;-21.1;9.8;47.5
---
> nli;-21.1;8.6;36.9
6497c6497
< non;-19.0;17.5;50.0
---
> non;-14.1;19.6;50.0
6565c6565
< nte;-20.5;10.3;39.4
---
> nte;-20.5;11.0;39.9
6619c6619
< nÑe;-13.4;16.9;49.3
---
> nÑe;-13.6;17.9;47.7
6644c6644
< o Ba;-18.1;14.1;44.6
---
> o Ba;-16.6;15.8;44.6
6661c6661
< o d;-16.1;15.6;48.8
---
> o d;-11.0;17.5;48.8
6670c6670
< o-s;-21.3;14.9;48.1
---
> o-s;-21.3;10.5;40.4
6676c6676
< oBa;-15.9;13.8;44.6
---
> oBa;-15.9;13.7;44.6
6683c6683
< oCan;-16.2;17.7;51.5
---
> oCan;-10.5;20.0;51.5
6708c6708
< oKai;-12.1;19.8;49.7
---
> oKai;-12.1;20.0;49.7
6747c6747
< oSaI;-11.5;19.3;51.4
---
> oSaI;-11.5;18.6;49.1
6777c6777
< oaT;-21.4;11.3;46.6
---
> oaT;-21.4;8.9;40.8
6783c6783
< oan;-17.2;15.1;43.9
---
> oan;-11.6;16.4;46.7
6790c6790
< oba;-16.6;16.9;47.9
---
> oba;-16.6;17.1;47.9
6825c6825
< oePo;-14.0;18.1;47.8
---
> oePo;-14.0;16.8;45.4
6896c6896
< oli;-20.5;17.2;47.5
---
> oli;-10.8;19.7;47.5
6953c6953
< onLu;-20.2;15.3;50.1
---
> onLu;-20.2;10.9;40.9
6999c6999
< ongo;-18.7;13.0;45.4
---
> ongo;-19.5;12.7;45.4
7041c7041
< ooke;-13.4;16.0;45.9
---
> ooke;-15.5;14.4;46.1
7043c7043
< oon;-23.8;11.1;48.6
---
> oon;-23.8;7.8;36.0
7048c7048
< opM;-14.5;14.6;43.1
---
> opM;-14.5;15.3;49.0
7099c7099
< oro;-17.4;11.0;39.2
---
> oro;-18.9;11.2;39.9
7166c7166
< ote;-21.0;9.0;39.9
---
> ote;-21.0;8.2;36.4
7288c7288
< pan;-22.5;11.3;43.4
---
> pan;-22.5;10.6;43.4
7292c7292
< para;-22.2;8.7;39.9
---
> para;-22.2;8.8;41.2
7346c7346
< pon;-23.1;10.2;42.2
---
> pon;-23.1;10.4;42.2
7359c7359
< pro;-20.6;10.0;40.7
---
> pro;-20.6;9.5;40.7
7393c7393
< q;-18.7;11.3;40.5
---
> q;-18.7;11.0;42.9
7409c7409
< r;-16.6;12.5;42.9
---
> r;-17.3;12.1;42.6
7453c7453
< rLeH;-11.8;18.1;47.0
---
> rLeH;-14.5;16.8;48.8
7480c7480
< rSvi;-21.9;14.2;49.8
---
> rSvi;-21.9;10.7;44.9
7492c7492
< ra;-15.9;14.9;46.4
---
> ra;-15.9;14.4;46.2
7499c7499
< raCh;-12.8;17.8;53.6
---
> raCh;-13.0;15.7;53.6
7510c7510
< raQ;-10.7;17.2;48.2
---
> raQ;-10.7;17.6;48.2
7586c7586
< rdS;-18.6;12.5;44.5
---
> rdS;-17.9;15.2;48.4
7615c7615
< red;-13.5;18.3;45.8
---
> red;-13.5;17.7;45.8
7622c7622
< ren;-20.3;17.5;51.7
---
> ren;-12.9;19.0;51.7
7644c7644
< rgBo;-20.1;12.2;45.2
---
> rgBo;-17.9;15.9;45.2
7694c7694
< rid;-9.6;18.8;47.2
---
> rid;-10.4;19.0;47.4
7707c7707
< ring;-10.4;19.6;53.1
---
> ring;-11.2;19.2;53.1
7720c7720
< riya;-22.2;9.8;43.2
---
> riya;-23.5;6.1;34.9
7724c7724
< riž;-19.5;13.0;41.3
---
> riž;-18.5;13.2;40.2
7774c7774
< rneK;-12.1;18.0;53.2
---
> rneK;-9.6;19.0;53.2
7860c7860
< rsNā;-24.0;13.0;47.0
---
> rsNā;-24.0;6.9;36.5
7896c7896
< ruTo;-12.0;19.9;50.8
---
> ruTo;-12.0;19.9;49.4
7962c7962
< s;-17.3;14.9;46.0
---
> s;-12.4;16.6;47.4
7984c7984
< s-d;-18.3;12.5;48.4
---
> s-d;-18.6;9.9;40.5
7998c7998
< sCa;-10.3;19.8;51.7
---
> sCa;-9.6;19.9;51.7
8060c8060
< sa;-17.6;14.9;46.2
---
> sa;-14.9;16.4;46.4
8076c8076
< sar;-18.9;12.5;43.6
---
> sar;-20.5;10.2;41.4
8108,8109c8108,8109
< sed;-13.4;19.2;50.6
< seki;-15.5;14.1;46.0
---
> sed;-10.4;19.9;50.6
> seki;-12.8;14.9;46.0
8118c8118
< set;-12.2;17.9;46.8
---
> set;-12.2;17.9;47.9
8161c8161
< sid;-10.4;18.5;48.0
---
> sid;-10.2;18.4;48.0
8164c8164
< sie;-12.0;19.7;52.7
---
> sie;-12.0;19.8;52.7
8215c8215
< sng;-10.6;17.3;48.4
---
> sng;-15.5;18.1;48.4
8221c8221
< soB;-22.2;10.3;42.4
---
> soB;-22.2;9.7;42.4
8242c8242
< sro;-20.6;9.8;41.2
---
> sro;-20.6;8.5;37.3
8320c8320
< t;-15.7;13.9;47.4
---
> t;-17.3;12.7;47.2
8355c8355
< tCa;-15.3;19.2;51.7
---
> tCa;-15.3;19.1;51.7
8397c8397
< ta;-22.3;9.6;44.6
---
> ta;-22.3;8.5;37.8
8450c8450
< tar;-20.5;15.4;50.7
---
> tar;-10.2;18.4;50.7
8497c8497
< ten;-19.8;9.7;44.6
---
> ten;-25.9;8.9;39.7
8548c8548
< tie;-16.3;19.7;52.7
---
> tie;-16.3;19.6;52.7
8579c8579
< tng;-15.5;15.9;48.0
---
> tng;-13.0;14.9;48.0
8583c8583
< to N;-17.0;14.2;52.4
---
> to N;-14.9;17.7;52.4
8586c8586
< toB;-19.9;11.3;43.8
---
> toB;-17.4;12.4;43.8
8645c8645
< tro;-18.7;11.9;41.8
---
> tro;-14.7;13.3;41.8
8704c8704
< u;-16.4;15.3;48.3
---
> u;-12.7;17.1;48.3
8757c8757
< ua;-19.9;17.3;47.1
---
> ua;-15.0;17.6;47.1
8764c8764
< uaQ;-18.8;12.8;44.8
---
> uaQ;-18.8;12.0;41.8
8778c8778
< uan;-11.1;19.2;48.7
---
> uan;-9.5;19.9;48.7
8811c8811
< ucaL;-17.7;13.9;44.9
---
> ucaL;-21.2;12.4;42.6
8857c8857
< uen;-25.9;8.4;39.7
---
> uen;-21.8;8.4;38.4
8885c8885
< uhSi;-21.8;16.7;50.9
---
> uhSi;-9.3;19.5;51.1
8921c8921
< ukka;-15.3;17.2;47.1
---
> ukka;-15.3;15.6;44.5
8987c8987
< ung;-14.8;14.5;44.7
---
> ung;-14.8;14.4;43.5
9023c9023
< urB;-13.4;15.6;47.0
---
> urB;-11.0;17.3;46.8
9045c9045
< urZa;-12.6;19.1;55.1
---
> urZa;-12.6;19.0;48.9
9066c9066
< uri;-25.3;13.1;43.6
---
> uri;-16.8;13.0;43.6
9076c9076
< uro;-14.8;15.8;45.2
---
> uro;-14.8;16.1;45.2
9172c9172
< v;-17.3;14.7;47.2
---
> v;-14.7;13.9;43.9
9190c9190
< van;-13.8;17.8;47.4
---
> van;-13.8;16.7;47.3
9247c9247
< vri;-19.8;13.4;42.4
---
> vri;-25.3;13.5;42.4
9257c9257
< w;-16.9;14.1;44.6
---
> w;-16.9;14.2;44.6
9317c9317
< x;-22.5;9.0;41.1
---
> x;-22.5;8.3;37.3
9328c9328
< xle;-22.5;11.6;45.9
---
> xle;-24.8;9.8;39.6
9335c9335
< y;-22.9;8.7;37.6
---
> y;-22.9;8.5;37.6
9441c9441
< yle;-24.8;15.0;50.3
---
> yle;-12.6;18.2;50.3
9486c9486
< z;-19.4;11.4;40.8
---
> z;-19.4;12.0;40.8
9738c9738
< ādS;-20.4;13.1;46.7
---
> ādS;-15.6;15.2;46.7
9785c9785
< ānp;-22.4;10.5;40.0
---
> ānp;-22.4;9.7;39.1
9808c9808
< ātk;-13.7;17.6;49.1
---
> ātk;-13.7;15.8;46.7
9857c9857
< īna;-13.0;17.8;51.1
---
> īna;-12.6;19.9;51.1
9902c9902
< Şır;-19.3;14.7;44.6
---
> Şır;-21.0;14.2;41.9
9924c9924
< ūKh;-11.2;19.6;48.3
---
> ūKh;-10.0;19.9;49.3

FAILURE: ./test.sh jonathan-aotearoa measurements_10K_1B.txt failed
```

**Comment by gunnarmorling** [comment]
Still seeing a (much smaller) diff for the 10K key set after the latest update:

```
Validating calculate_average_jonathan-aotearoa.sh -- measurements_10K_1B.txt
Using native image 'target/CalculateAverage_jonathan-aotearoa_image'. Delete this file to select JVM mode.
234c234
< Bālu;-18.3;15.0;50.9
---
> Bālu;-8.4;19.8;52.0
982c982
< Pālg;-23.5;15.0;52.0
---
> Pālg;-23.5;10.2;41.0

FAILURE: ./test.sh jonathan-aotearoa measurements_10K_1B.txt failed
Summary
  jonathan-aotearoa: command failed or output did not match
```

**Comment by jonathan-aotearoa** [comment]
@gunnarmorling - Apologies, I had a bug in my rounding code. Should be OK now. I've tested it against 10k versions of `measurements2.txt` and `measurements3.txt`

**Comment by jonathan-aotearoa** [comment]
Output from my `TestRunner`

```
Testing 'measurements-10000-unique-keys.txt'... Passed
Testing 'measurements-boundaries.txt'... Passed
Testing 'measurements-rounding.txt'... Passed
Testing 'measurements-dot.txt'... Passed
Testing 'measurements-3-10k.txt'... Passed
Testing 'measurements-short.txt'... Passed
Testing 'measurements-10.txt'... Passed
Testing 'measurements-2-10k.txt'... Passed
Testing 'measurements-complex-utf8.txt'... Passed
Testing 'measurements-2.txt'... Passed
Testing 'measurements-shortest.txt'... Passed
Testing 'measurements-3.txt'... Passed
Testing 'measurements-20.txt'... Passed
Testing 'measurements-1.txt'... Passed
```

**Comment by gunnarmorling** [comment]
Still failing:

```
Validating calculate_average_jonathan-aotearoa.sh -- measurements_10K_1B.txt
Using native image 'target/CalculateAverage_jonathan-aotearoa_image'. Delete this file to select JVM mode.
234c234
< Bālu;-18.3;15.0;50.9
---
> Bālu;-8.4;19.8;52.0
982c982
< Pālg;-23.5;15.0;52.0
---
> Pālg;-23.5;10.2;41.0

FAILURE: ./test.sh jonathan-aotearoa measurements_10K_1B.txt failed
Summary
  jonathan-aotearoa: command failed or output did not match
```

Note how max/min are off, can't just be rounding.

**Comment by jonathan-aotearoa** [comment]
This is really strange. I'm generating data using `./create_measurements3.sh 10000` and comparing my output with the output from `CalculateAverage_baseline`. I've tried is several times just now and it's always identical.

**Comment by jonathan-aotearoa** [comment]
Is there another command I can run to generate data similar to the build?

**Comment by gunnarmorling** [comment]
I reckon it's caused by running the tests with 32 cores on the eval machine
which may somehow trip up the partitioning logic.

**Comment by jonathan-aotearoa** [comment]
OK, I see the problem. I still have an issue with my collision checking. I just generated a 200k line file and I got 4 mismatches.

**Comment by jonathan-aotearoa** [comment]
`./create_measurements3.sh 100000` is now passing

**Comment by gunnarmorling** [comment]
Yepp, all good now.  00:05.077 on the standard key set and the 10K one passes to (in 00:07.499).

**Pull request merged**